### PR TITLE
Use winapi

### PR DESCRIPTION
**Pull request by retep998**
Also accidentally removed trailing whitespace.

### CODE REVIEW

**Comment by seanmonstar** [comment]
Thanks!

Previously, num_cpus did actually depend on winapi, and @alexcrichton sent a [PR removing it](https://github.com/seanmonstar/num_cpus/pull/20). I believe the rationale was that num_cpus only needs 2 funtion definitions, and so downloading and compile winapi negatively impacts builds, and it's pretty easy to just have these functions defined internally.

Is there a reason to revert that?

**Comment by retep998** [comment]
The issue was that `winapi` 0.2 took a long time to compile (over 23 seconds on a Ryzen). `winapi` 0.3 meanwhile is much quicker. With this PR the total build time for me  goes from 0.5 seconds to 3.2 seconds so while not perfect, it is much better than when `winapi` was removed.. Making `libc` conditional on `not(windows)` also saves 298KiB which offsets the 884KiB download of `winapi` (which makes me wonder why you *don't* care about the impact of `libc` on build times when it is **unused** on Windows).

So basically it's up to you to decide whether `winapi`'s compile times have improved enough for it to be considered an acceptable dependency.

**Comment by newpavlov** [comment]
I believe it should be enough to just add `[target.'cfg(not(windows))'.dependencies]` for `libc`, while depending on the whole `winapi` just for one function and struct in my opinion is simply not reasonable.
